<a href="https://colab.research.google.com/github/simmy4946/NEW-REPOSITS/blob/main/Walmart_%26_Amazon.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Import Libararies:**

In [ ]:
import pandas as pd

**Import Data:**

In [ ]:
df = pd.read_excel("/content/walmart_amazon_final.xlsx")

**see top 5 heads of data**

In [ ]:
df.head(5)

WALMART_NAME  \
0  Easter Bunny Wood Door Sign - Home Decor - 1 P...   
1  Kinder Bueno Milk Chocolate and Hazelnut Cream...   
2  8" Brown Ivory and Orange Standing Duck Spring...   
3  Caffoz 58" Home Entertainment Center Wood Medi...   
4            Novogratz Voler Coffee Table, Light Oak   

                                         WALMART_URL  \
0  https://www.walmart.com/ip/Easter-Bunny-Wood-D...   
1  https://www.walmart.com/ip/Kinder-Bueno-Milk-C...   
2  https://www.walmart.com/ip/8-Brown-Ivory-and-O...   
3  https://www.walmart.com/ip/Caffoz-58-Home-Ente...   
4  https://www.walmart.com/ip/Novogratz-Voler-Cof...   

                                         AMAZON_NAME  \
0  Easter Bunny Wood Door Sign - Home Decor - 1 P...   
1  Kinder Bueno Milk Chocolate and Hazelnut Cream...   
2  8" Brown Ivory and Orange Standing Duck Spring...   
3  Caffoz 58" Home Entertainment Center Wood Medi...   
4            Novogratz Voler Coffee Table, Light Oak   

                             AMAZON_URL  WALMART_PRICE_CURRENT  \
0  https://www.amazon.com/dp/B083MQKGSR                  14.29   
1  https://www.amazon.com/dp/B08Z7J19JR                   3.58   
2  https://www.amazon.com/dp/B07YMBP113                  18.99   
3  https://www.amazon.com/dp/B0BGYW1BDD                 109.78   
4  https://www.amazon.com/dp/B0BR65517B                 106.46   

   AMAZON_PRICE_CURRENT  SCORE                    DEPARTMENT          CATEGORY  
0                 15.57    1.0                        Easter  Easter Top Picks  
1                  3.58    1.0                        Easter  Easter Top Picks  
2                 18.99    1.0                        Easter  Easter Top Picks  
3                142.72    1.0  Home, Furniture & Appliances         Furniture  
4                124.20    1.0  Home, Furniture & Appliances         Furniture

**see the shape of data:**

In [ ]:
df.shape

(51, 9)

**check the columns of data:**

In [ ]:
df.columns

Index(['WALMART_NAME', 'WALMART_URL', 'AMAZON_NAME', 'AMAZON_URL',
       'WALMART_PRICE_CURRENT', 'AMAZON_PRICE_CURRENT', 'SCORE', 'DEPARTMENT',
       'CATEGORY'],
      dtype='object')

**drop the columns which we don't need**

In [ ]:
df.drop(['WALMART_URL', 'AMAZON_URL'], axis=1,inplace=True)
df.head(2)

WALMART_NAME  \
0  Easter Bunny Wood Door Sign - Home Decor - 1 P...   
1  Kinder Bueno Milk Chocolate and Hazelnut Cream...   

                                         AMAZON_NAME  WALMART_PRICE_CURRENT  \
0  Easter Bunny Wood Door Sign - Home Decor - 1 P...                  14.29   
1  Kinder Bueno Milk Chocolate and Hazelnut Cream...                   3.58   

   AMAZON_PRICE_CURRENT  SCORE DEPARTMENT          CATEGORY  
0                 15.57    1.0     Easter  Easter Top Picks  
1                  3.58    1.0     Easter  Easter Top Picks

**see the null values in the data**

In [ ]:
df.isnull().sum()

WALMART_NAME             1
AMAZON_NAME              2
WALMART_PRICE_CURRENT    2
AMAZON_PRICE_CURRENT     2
SCORE                    2
DEPARTMENT               2
CATEGORY                 2
dtype: int64

there is, some missing or null values in the columns which we need to require treatment for them.

**we treated null values here by filling the values with mean**

In [ ]:
df['WALMART_PRICE_CURRENT'].fillna(df['WALMART_PRICE_CURRENT'].mean(), inplace=True)
df['AMAZON_PRICE_CURRENT'].fillna(df['AMAZON_PRICE_CURRENT'].mean(), inplace=True)
df['SCORE'].fillna(df['SCORE'].mean(), inplace=True)


In [ ]:
mode_value = df['DEPARTMENT'].mode()[0]
df['DEPARTMENT'].fillna(mode_value, inplace=True)
mode_value = df['DEPARTMENT'].mode()[0]
df['CATEGORY'].fillna(mode_value, inplace=True)
mode_value = df['WALMART_NAME'].mode()[0]
df['WALMART_NAME'].fillna(mode_value, inplace=True)
mode_value = df['AMAZON_NAME'].mode()[0]
df['AMAZON_NAME'].fillna(mode_value, inplace=True)


**check the null values again after treating them**

In [ ]:
df.isnull().sum()

WALMART_NAME             0
AMAZON_NAME              0
WALMART_PRICE_CURRENT    0
AMAZON_PRICE_CURRENT     0
SCORE                    0
DEPARTMENT               0
CATEGORY                 0
dtype: int64

In [ ]:
#drops_columns=['WALMART_URL','AMAZON_URL']
#df.drop(columns=drops_columns, axis=1, inplace=True)
#df.head(2) # inplace =True (must be there otw code will not work)

**How many unique products are listed on Walmart and Amazon?**

In [ ]:
df['AMAZON_NAME'].nunique()

49

there is 49 unique names in 'AMAZON_NAME' column

In [ ]:
df['WALMART_NAME'].nunique()

50

In [ ]:
df.head(1)

WALMART_NAME  \
0  Easter Bunny Wood Door Sign - Home Decor - 1 P...   

                                         AMAZON_NAME  WALMART_PRICE_CURRENT  \
0  Easter Bunny Wood Door Sign - Home Decor - 1 P...                  14.29   

   AMAZON_PRICE_CURRENT  SCORE DEPARTMENT          CATEGORY  
0                 15.57    1.0     Easter  Easter Top Picks

there is 50 unique names in 'WALMART_NAME' column

**what is the total sum of prices for each unique 'WALMART_NAME' and 'AMAZON_NAME**

In [ ]:
df.groupby('WALMART_NAME')['WALMART_PRICE_CURRENT'].sum().reset_index().sort_values(by='WALMART_PRICE_CURRENT' )


WALMART_NAME  WALMART_PRICE_CURRENT
35  McCormick Air Fryer Loaded Seasoning Mix, 1.25 oz               1.480000
31  Kinder Bueno Milk Chocolate and Hazelnut Cream...               3.580000
14                     Coghlan's Featherweight Mirror               6.670000
25  Evriholder Lemon and Lime Squeezer, Yellow or ...               7.000000
42  Ovente French Coffee Press 12 Ounce Espresso M...              11.990000
27  Gallery Solutions 8x10 Black Wall or Tabletop ...              14.200000
24  Easter Bunny Wood Door Sign - Home Decor - 1 P...              14.290000
1   10 Strawberry Street Two Tone Coupe 5.5?€?/24 ...              16.910000
47  Smart Design Undershelf Storage Basket - Small...              17.490000
7   Americanflat 12x18 Poster Frame in Silver - Co...              17.620000
3   8" Brown Ivory and Orange Standing Duck Spring...              18.990000
36  Mind Reader 3 Tier Letter Tray Document Sorter...              19.590000
2   10 Strawberry Street Two Tone Coupe 8.5?€?/36 ...              24.490000
11  Arden Selections Outdoor Toss Pillow (2 Pack) ...              24.990000
5   Alera ALEFW601416BL Compact 15.25 in. x 12.38 ...              26.110000
26       Furinno Simple Design Coffee Table, Espresso              27.760000
33  Lavish Home 4-Tier Wood Folding Corner Display...              28.030000
43  Patton Wall Decor 20" Black and Gold Metal Cut...              30.830000
19  Convenience Concepts Designs2Go 4 Tier Corner ...              31.980000
4   AcuRite Intelli-Time Digital Alarm Clock for B...              32.790000
8   Americanflat 20x26 Puzzle Frame with Mat for 1...              34.990000
6   Alera ALEFW601424BL 24 in. x 14 in. x 21 in. S...              36.050000
32  KitchenAid Variable Speed Corded Hand Blender ...              50.990000
22  Convenience Concepts Xtra Storage 5 Tier Foldi...              53.540000
29  Instapoints 5 Piece Gallery Wall Picture Frame...              54.440000
41  Ovente Air Fryer 3.2 QT, 1400W with Adjustable...              55.980000
46  Signature Design by Ashley Media Modern Geomet...              59.650000
20  Convenience Concepts Designs2Go Highboy TV Sta...              89.000000
40  Ovente 8.5'' Hardwired Lighted Wall Mount Make...              89.940000
15       Contact US to Get 5000+ products comparisons              97.069388
16  Convenience Concepts American Heritage Bookshe...             105.200000
21  Convenience Concepts Oxford Coffee Table with ...             105.580000
39            Novogratz Voler Coffee Table, Light Oak             106.460000
13  Caffoz 58" Home Entertainment Center Wood Medi...             109.780000
17  Convenience Concepts American Heritage Coffee ...             117.990000
0   10 Strawberry Street Simply White 36 Oz Pasta/...             119.059388
18  Convenience Concepts American Heritage Round C...             127.000000
38        Novogratz Kelly Accent Cabinet, White/Taupe             133.750000
28  GreenPan Swift Healthy Ceramic Nonstick, 12 Pi...             134.990000
10  Ameriwood Home Pursuit Mobile File Cabinet, Ch...             145.230000
45  Sceptre 40" Class FHD (1080P) LED TV (X405BV-FSR)             148.000000
23         DHP Jett Junior Twin Metal Loft Bed, Black             150.000000
30  Kate and Laurel Cates Farmhouse Wood Framed Wa...             157.750000
37  Novogratz Kelly 5 Drawer Dresser, Faux Rattan ...             170.770000
44  Prepac Space-Saving Entryway Organizer with Sh...             178.990000
12               Boahaus Alva Dresser (Brown - White)             180.990000
48      South Shore Fusion 5-Drawer Chest, Pure White             199.990000
49  South Shore Step One Essential 5-Drawer Chest,...             201.990000
9       Ameriwood Home Lamont Mantel Fireplace, White             330.390000
34  Lenovo ThinkCentre M70a Gen 3 11VL0040US Deskt...            1028.190000

the total sum of the prices of each unique product the wallmart is starts from 1.48 to ends with 1028.19 rupees. McCormick Air Fryer Loaded Seasoning Mix, 1.25 oz	products is the lowest prices.

In [ ]:
df.groupby('AMAZON_NAME')['AMAZON_PRICE_CURRENT'].sum().reset_index().sort_values(by='AMAZON_PRICE_CURRENT' )

AMAZON_NAME  AMAZON_PRICE_CURRENT
34  McCormick Air Fryer Loaded Seasoning Mix, 1.25 oz              1.490000
30  Kinder Bueno Milk Chocolate and Hazelnut Cream...              3.580000
14                     Coghlan's Featherweight Mirror              6.170000
24  Evriholder Lemon and Lime Squeezer, Yellow or ...              9.990000
46  Smart Design Undershelf Storage Basket - Small...             11.990000
41  Ovente French Coffee Press 12 Ounce Espresso M...             11.990000
26  Gallery Solutions 8x10 Black Wall or Tabletop ...             14.200000
23  Easter Bunny Wood Door Sign - Home Decor - 1 P...             15.570000
1   10 Strawberry Street Two Tone Coupe 5.5?€?/24 ...             16.910000
7   Americanflat 12x18 Poster Frame in Silver - Co...             17.620000
3   8" Brown Ivory and Orange Standing Duck Spring...             18.990000
35  Mind Reader 3 Tier Letter Tray Document Sorter...             19.590000
2   10 Strawberry Street Two Tone Coupe 8.5?€?/36 ...             24.490000
11  Arden Selections Outdoor Toss Pillow (2 Pack) ...             24.990000
42  Patton Wall Decor 20" Black and Gold Metal Cut...             26.990000
5   Alera ALEFW601416BL Compact 15.25 in. x 12.38 ...             27.610000
32  Lavish Home 4-Tier Wood Folding Corner Display...             27.950000
25       Furinno Simple Design Coffee Table, Espresso             29.770000
4   AcuRite Intelli-Time Digital Alarm Clock for B...             32.790000
18  Convenience Concepts Designs2Go 4 Tier Corner ...             33.990000
8   Americanflat 20x26 Puzzle Frame with Mat for 1...             34.990000
6   Alera ALEFW601424BL 24 in. x 14 in. x 21 in. S...             37.670000
31  KitchenAid Variable Speed Corded Hand Blender ...             50.990000
28  Instapoints 5 Piece Gallery Wall Picture Frame...             54.440000
40  Ovente Air Fryer 3.2 QT, 1400W with Adjustable...             55.980000
21  Convenience Concepts Xtra Storage 5 Tier Foldi...             59.490000
45  Signature Design by Ashley Media Modern Geomet...             59.650000
19  Convenience Concepts Designs2Go Highboy TV Sta...             89.000000
39  Ovente 8.5'' Hardwired Lighted Wall Mount Make...             89.940000
15  Convenience Concepts American Heritage Bookshe...            105.200000
20  Convenience Concepts Oxford Coffee Table with ...            105.580000
16  Convenience Concepts American Heritage Coffee ...            117.990000
38            Novogratz Voler Coffee Table, Light Oak            124.200000
17  Convenience Concepts American Heritage Round C...            127.000000
37        Novogratz Kelly Accent Cabinet, White/Taupe            133.750000
27  GreenPan Swift Healthy Ceramic Nonstick, 12 Pi...            134.990000
13  Caffoz 58" Home Entertainment Center Wood Medi...            142.720000
10  Ameriwood Home Pursuit Mobile File Cabinet, Ch...            145.230000
22         DHP Jett Junior Twin Metal Loft Bed, Black            150.000000
29  Kate and Laurel Cates Farmhouse Wood Framed Wa...            157.750000
36  Novogratz Kelly 5 Drawer Dresser, Faux Rattan ...            170.770000
43  Prepac Space-Saving Entryway Organizer with Sh...            178.990000
12               Boahaus Alva Dresser (Brown - White)            180.990000
47      South Shore Fusion 5-Drawer Chest, Pure White            199.990000
48  South Shore Step One Essential 5-Drawer Chest,...            201.370000
44  Sceptre 40" Class FHD (1080P) LED TV (X405BV-FSR)            207.000000
0   10 Strawberry Street Simply White 36 Oz Pasta/...            220.687551
9       Ameriwood Home Lamont Mantel Fireplace, White            330.390000
33  Lenovo ThinkCentre M70a Gen 3 11VL0040US Deskt...           1023.370000

the total sum of the prices of each unique product the amazon is starts from 1.49 to ends with 1023.19 rupees. Lenovo ThinkCentre M70a Gen 3 11VL0040US Desktop is the products with highest prices

In [ ]:
df.groupby('AMAZON_NAME')['AMAZON_PRICE_CURRENT'].sum().reset_index().sort_values(by='AMAZON_PRICE_CURRENT')


AMAZON_NAME  AMAZON_PRICE_CURRENT
34  McCormick Air Fryer Loaded Seasoning Mix, 1.25 oz              1.490000
30  Kinder Bueno Milk Chocolate and Hazelnut Cream...              3.580000
14                     Coghlan's Featherweight Mirror              6.170000
24  Evriholder Lemon and Lime Squeezer, Yellow or ...              9.990000
46  Smart Design Undershelf Storage Basket - Small...             11.990000
41  Ovente French Coffee Press 12 Ounce Espresso M...             11.990000
26  Gallery Solutions 8x10 Black Wall or Tabletop ...             14.200000
23  Easter Bunny Wood Door Sign - Home Decor - 1 P...             15.570000
1   10 Strawberry Street Two Tone Coupe 5.5?€?/24 ...             16.910000
7   Americanflat 12x18 Poster Frame in Silver - Co...             17.620000
3   8" Brown Ivory and Orange Standing Duck Spring...             18.990000
35  Mind Reader 3 Tier Letter Tray Document Sorter...             19.590000
2   10 Strawberry Street Two Tone Coupe 8.5?€?/36 ...             24.490000
11  Arden Selections Outdoor Toss Pillow (2 Pack) ...             24.990000
42  Patton Wall Decor 20" Black and Gold Metal Cut...             26.990000
5   Alera ALEFW601416BL Compact 15.25 in. x 12.38 ...             27.610000
32  Lavish Home 4-Tier Wood Folding Corner Display...             27.950000
25       Furinno Simple Design Coffee Table, Espresso             29.770000
4   AcuRite Intelli-Time Digital Alarm Clock for B...             32.790000
18  Convenience Concepts Designs2Go 4 Tier Corner ...             33.990000
8   Americanflat 20x26 Puzzle Frame with Mat for 1...             34.990000
6   Alera ALEFW601424BL 24 in. x 14 in. x 21 in. S...             37.670000
31  KitchenAid Variable Speed Corded Hand Blender ...             50.990000
28  Instapoints 5 Piece Gallery Wall Picture Frame...             54.440000
40  Ovente Air Fryer 3.2 QT, 1400W with Adjustable...             55.980000
21  Convenience Concepts Xtra Storage 5 Tier Foldi...             59.490000
45  Signature Design by Ashley Media Modern Geomet...             59.650000
19  Convenience Concepts Designs2Go Highboy TV Sta...             89.000000
39  Ovente 8.5'' Hardwired Lighted Wall Mount Make...             89.940000
15  Convenience Concepts American Heritage Bookshe...            105.200000
20  Convenience Concepts Oxford Coffee Table with ...            105.580000
16  Convenience Concepts American Heritage Coffee ...            117.990000
38            Novogratz Voler Coffee Table, Light Oak            124.200000
17  Convenience Concepts American Heritage Round C...            127.000000
37        Novogratz Kelly Accent Cabinet, White/Taupe            133.750000
27  GreenPan Swift Healthy Ceramic Nonstick, 12 Pi...            134.990000
13  Caffoz 58" Home Entertainment Center Wood Medi...            142.720000
10  Ameriwood Home Pursuit Mobile File Cabinet, Ch...            145.230000
22         DHP Jett Junior Twin Metal Loft Bed, Black            150.000000
29  Kate and Laurel Cates Farmhouse Wood Framed Wa...            157.750000
36  Novogratz Kelly 5 Drawer Dresser, Faux Rattan ...            170.770000
43  Prepac Space-Saving Entryway Organizer with Sh...            178.990000
12               Boahaus Alva Dresser (Brown - White)            180.990000
47      South Shore Fusion 5-Drawer Chest, Pure White            199.990000
48  South Shore Step One Essential 5-Drawer Chest,...            201.370000
44  Sceptre 40" Class FHD (1080P) LED TV (X405BV-FSR)            207.000000
0   10 Strawberry Street Simply White 36 Oz Pasta/...            220.687551
9       Ameriwood Home Lamont Mantel Fireplace, White            330.390000
33  Lenovo ThinkCentre M70a Gen 3 11VL0040US Deskt...           1023.370000

**How does Walmart pricing compare to Amazon pricing across different departments or categories?**

In [ ]:
amazon_df=df[['AMAZON_NAME','DEPARTMENT','AMAZON_PRICE_CURRENT']]

In [ ]:
walmart_df=df[['WALMART_NAME','DEPARTMENT','WALMART_PRICE_CURRENT']]

In [ ]:
merged_df = pd.merge(walmart_df, amazon_df, on=['DEPARTMENT'], how='inner')
merged_df

WALMART_NAME  \
0     Easter Bunny Wood Door Sign - Home Decor - 1 P...   
1     Easter Bunny Wood Door Sign - Home Decor - 1 P...   
2     Easter Bunny Wood Door Sign - Home Decor - 1 P...   
3     Kinder Bueno Milk Chocolate and Hazelnut Cream...   
4     Kinder Bueno Milk Chocolate and Hazelnut Cream...   
...                                                 ...   
2214       Contact US to Get 5000+ products comparisons   
2215       Contact US to Get 5000+ products comparisons   
2216       Contact US to Get 5000+ products comparisons   
2217       Contact US to Get 5000+ products comparisons   
2218  Lenovo ThinkCentre M70a Gen 3 11VL0040US Deskt...   

                        DEPARTMENT  WALMART_PRICE_CURRENT  \
0                           Easter              14.290000   
1                           Easter              14.290000   
2                           Easter              14.290000   
3                           Easter               3.580000   
4                           Easter               3.580000   
...                            ...                    ...   
2214  Home, Furniture & Appliances              97.069388   
2215  Home, Furniture & Appliances              97.069388   
2216  Home, Furniture & Appliances              97.069388   
2217  Home, Furniture & Appliances              97.069388   
2218                   Electronics            1028.190000   

                                            AMAZON_NAME  AMAZON_PRICE_CURRENT  
0     Easter Bunny Wood Door Sign - Home Decor - 1 P...             15.570000  
1     Kinder Bueno Milk Chocolate and Hazelnut Cream...              3.580000  
2     8" Brown Ivory and Orange Standing Duck Spring...             18.990000  
3     Easter Bunny Wood Door Sign - Home Decor - 1 P...             15.570000  
4     Kinder Bueno Milk Chocolate and Hazelnut Cream...              3.580000  
...                                                 ...                   ...  
2214  Smart Design Undershelf Storage Basket - Small...             11.990000  
2215  Convenience Concepts American Heritage Bookshe...            105.200000  
2216  10 Strawberry Street Simply White 36 Oz Pasta/...             99.348776  
2217  10 Strawberry Street Simply White 36 Oz Pasta/...             99.348776  
2218  Lenovo ThinkCentre M70a Gen 3 11VL0040US Deskt...           1023.370000  

[2219 rows x 5 columns]

from above the comprises and comparison between the walmart and amazon prices with their respective departments, it shows the common departments with their mention prices.

**Are there any pricing trends or patterns within specific categories or departments?**

In [ ]:
df.groupby('DEPARTMENT')[['WALMART_PRICE_CURRENT','AMAZON_PRICE_CURRENT']].mean().sort_values(by='WALMART_PRICE_CURRENT')

WALMART_PRICE_CURRENT  AMAZON_PRICE_CURRENT
DEPARTMENT                                                               
Easter                                    12.286667             12.713333
Home, Furniture & Appliances              82.669974             85.218671
Electronics                             1028.190000           1023.370000

In [ ]:
df.groupby('CATEGORY')[['WALMART_PRICE_CURRENT','AMAZON_PRICE_CURRENT']].mean().sort_values(by='WALMART_PRICE_CURRENT')

WALMART_PRICE_CURRENT  AMAZON_PRICE_CURRENT
CATEGORY                                                                 
Easter Top Picks                          12.286667             12.713333
Bath                                      22.760000             19.970000
Kitchen & Dining                          40.542500             40.916250
Decor                                     46.422000             45.988000
Storage & Organization                    52.083333             53.930000
Bedding                                   59.650000             59.650000
Home, Furniture & Appliances              97.069388             99.348776
Home                                      99.826667            119.496667
Kid's Rooms                              105.200000            105.200000
Furniture                                134.619231            138.624615
Appliances                               330.390000            330.390000
Computers                               1028.190000           1023.370000

How many products are common between Walmart and Amazon?
What are the top departments and categories represented in the dataset?

What are the top departments and categories represented in the dataset?



In [ ]:
df[['DEPARTMENT', 'CATEGORY']].value_counts().head(5).reset_index()

DEPARTMENT                CATEGORY   0
0  Home, Furniture & Appliances               Furniture  13
1  Home, Furniture & Appliances                   Decor  10
2  Home, Furniture & Appliances        Kitchen & Dining   8
3  Home, Furniture & Appliances  Storage & Organization   6
4                        Easter        Easter Top Picks   3

What is the average, minimum, and maximum price of Walmart products?

In [ ]:
#df.groupby('WALMART_NAME')['WALMART_PRICE_CURRENT'].mean().reset_index()

In [ ]:
df.groupby('WALMART_NAME').agg({'WALMART_PRICE_CURRENT':['mean', 'median','max','min']}).reset_index()

WALMART_NAME WALMART_PRICE_CURRENT  \
                                                                       mean   
0   10 Strawberry Street Simply White 36 Oz Pasta/...             59.529694   
1   10 Strawberry Street Two Tone Coupe 5.5?€?/24 ...             16.910000   
2   10 Strawberry Street Two Tone Coupe 8.5?€?/36 ...             24.490000   
3   8" Brown Ivory and Orange Standing Duck Spring...             18.990000   
4   AcuRite Intelli-Time Digital Alarm Clock for B...             32.790000   
5   Alera ALEFW601416BL Compact 15.25 in. x 12.38 ...             26.110000   
6   Alera ALEFW601424BL 24 in. x 14 in. x 21 in. S...             36.050000   
7   Americanflat 12x18 Poster Frame in Silver - Co...             17.620000   
8   Americanflat 20x26 Puzzle Frame with Mat for 1...             34.990000   
9       Ameriwood Home Lamont Mantel Fireplace, White            330.390000   
10  Ameriwood Home Pursuit Mobile File Cabinet, Ch...            145.230000   
11  Arden Selections Outdoor Toss Pillow (2 Pack) ...             24.990000   
12               Boahaus Alva Dresser (Brown - White)            180.990000   
13  Caffoz 58" Home Entertainment Center Wood Medi...            109.780000   
14                     Coghlan's Featherweight Mirror              6.670000   
15       Contact US to Get 5000+ products comparisons             97.069388   
16  Convenience Concepts American Heritage Bookshe...            105.200000   
17  Convenience Concepts American Heritage Coffee ...            117.990000   
18  Convenience Concepts American Heritage Round C...            127.000000   
19  Convenience Concepts Designs2Go 4 Tier Corner ...             31.980000   
20  Convenience Concepts Designs2Go Highboy TV Sta...             89.000000   
21  Convenience Concepts Oxford Coffee Table with ...            105.580000   
22  Convenience Concepts Xtra Storage 5 Tier Foldi...             53.540000   
23         DHP Jett Junior Twin Metal Loft Bed, Black            150.000000   
24  Easter Bunny Wood Door Sign - Home Decor - 1 P...             14.290000   
25  Evriholder Lemon and Lime Squeezer, Yellow or ...              7.000000   
26       Furinno Simple Design Coffee Table, Espresso             27.760000   
27  Gallery Solutions 8x10 Black Wall or Tabletop ...             14.200000   
28  GreenPan Swift Healthy Ceramic Nonstick, 12 Pi...            134.990000   
29  Instapoints 5 Piece Gallery Wall Picture Frame...             54.440000   
30  Kate and Laurel Cates Farmhouse Wood Framed Wa...            157.750000   
31  Kinder Bueno Milk Chocolate and Hazelnut Cream...              3.580000   
32  KitchenAid Variable Speed Corded Hand Blender ...             50.990000   
33  Lavish Home 4-Tier Wood Folding Corner Display...             28.030000   
34  Lenovo ThinkCentre M70a Gen 3 11VL0040US Deskt...           1028.190000   
35  McCormick Air Fryer Loaded Seasoning Mix, 1.25 oz              1.480000   
36  Mind Reader 3 Tier Letter Tray Document Sorter...             19.590000   
37  Novogratz Kelly 5 Drawer Dresser, Faux Rattan ...            170.770000   
38        Novogratz Kelly Accent Cabinet, White/Taupe            133.750000   
39            Novogratz Voler Coffee Table, Light Oak            106.460000   
40  Ovente 8.5'' Hardwired Lighted Wall Mount Make...             89.940000   
41  Ovente Air Fryer 3.2 QT, 1400W with Adjustable...             55.980000   
42  Ovente French Coffee Press 12 Ounce Espresso M...             11.990000   
43  Patton Wall Decor 20" Black and Gold Metal Cut...             30.830000   
44  Prepac Space-Saving Entryway Organizer with Sh...            178.990000   
45  Sceptre 40" Class FHD (1080P) LED TV (X405BV-FSR)            148.000000   
46  Signature Design by Ashley Media Modern Geomet...             59.650000   
47  Smart Design Undershelf Storage Basket - Small...             17.490000   
48      South Shore Fusion 5-Drawer Chest, Pure White            199.990000   
49  South Sh

Is there any noticeable price difference between Walmart and Amazon for the same products?

In [ ]:
df1=df[['WALMART_PRICE_CURRENT','AMAZON_PRICE_CURRENT']]
df1

WALMART_PRICE_CURRENT  AMAZON_PRICE_CURRENT
0               14.290000             15.570000
1                3.580000              3.580000
2               18.990000             18.990000
3              109.780000            142.720000
4              106.460000            124.200000
5               27.760000             29.770000
6              105.580000            105.580000
7                1.480000              1.490000
8              148.000000            207.000000
9               89.000000             89.000000
10             150.000000            150.000000
11             127.000000            127.000000
12             117.990000            117.990000
13             178.990000            178.990000
14             180.990000            180.990000
15              55.980000             55.980000
16             133.750000            133.750000
17             170.770000            170.770000
18             134.990000            134.990000
19              16.910000             16.910000
20              24.490000             24.490000
21             199.990000            199.990000
22             201.990000            201.370000
23              21.990000             21.990000
24              31.980000             33.990000
25              53.540000             59.490000
26              50.990000             50.990000
27              19.590000             19.590000
28              11.990000             11.990000
29              26.110000             27.610000
30             145.230000            145.230000
31               7.000000              9.990000
32              36.050000             37.670000
33              32.790000             32.790000
34              54.440000             54.440000
35              24.990000             24.990000
36             157.750000            157.750000
37              14.200000             14.200000
38             330.390000            330.390000
39              89.940000             89.940000
40               6.670000              6.170000
41              30.830000             26.990000
42              59.650000             59.650000
43              34.990000             34.990000
44              17.620000             17.620000
45              28.030000             27.950000
46              17.490000             11.990000
47            1028.190000           1023.370000
48             105.200000            105.200000
49              97.069388             99.348776
50              97.069388             99.348776

In [ ]:
df['DIFFERENCE_OF_PRICES']=df['WALMART_PRICE_CURRENT']-df['AMAZON_PRICE_CURRENT']

In [ ]:
df2=df['DIFFERENCE_OF_PRICES']
df2.reset_index()

index  DIFFERENCE_OF_PRICES
0       0             -1.280000
1       1              0.000000
2       2              0.000000
3       3            -32.940000
4       4            -17.740000
5       5             -2.010000
6       6              0.000000
7       7             -0.010000
8       8            -59.000000
9       9              0.000000
10     10              0.000000
11     11              0.000000
12     12              0.000000
13     13              0.000000
14     14              0.000000
15     15              0.000000
16     16              0.000000
17     17              0.000000
18     18              0.000000
19     19              0.000000
20     20              0.000000
21     21              0.000000
22     22              0.620000
23     23              0.000000
24     24             -2.010000
25     25             -5.950000
26     26              0.000000
27     27              0.000000
28     28              0.000000
29     29             -1.500000
30     30              0.000000
31     31             -2.990000
32     32             -1.620000
33     33              0.000000
34     34              0.000000
35     35              0.000000
36     36              0.000000
37     37              0.000000
38     38              0.000000
39     39              0.000000
40     40              0.500000
41     41              3.840000
42     42              0.000000
43     43              0.000000
44     44              0.000000
45     45              0.080000
46     46              5.500000
47     47              4.820000
48     48              0.000000
49     49             -2.279388
50     50             -2.279388

In [ ]:
final_df = pd.concat([df1, df2], axis=1) # row wise concate
final_df
#merged_df = pd.merge(walmart_df, amazon_df, on='Product')


WALMART_PRICE_CURRENT  AMAZON_PRICE_CURRENT  DIFFERENCE_OF_PRICES
0               14.290000             15.570000             -1.280000
1                3.580000              3.580000              0.000000
2               18.990000             18.990000              0.000000
3              109.780000            142.720000            -32.940000
4              106.460000            124.200000            -17.740000
5               27.760000             29.770000             -2.010000
6              105.580000            105.580000              0.000000
7                1.480000              1.490000             -0.010000
8              148.000000            207.000000            -59.000000
9               89.000000             89.000000              0.000000
10             150.000000            150.000000              0.000000
11             127.000000            127.000000              0.000000
12             117.990000            117.990000              0.000000
13             178.990000            178.990000              0.000000
14             180.990000            180.990000              0.000000
15              55.980000             55.980000              0.000000
16             133.750000            133.750000              0.000000
17             170.770000            170.770000              0.000000
18             134.990000            134.990000              0.000000
19              16.910000             16.910000              0.000000
20              24.490000             24.490000              0.000000
21             199.990000            199.990000              0.000000
22             201.990000            201.370000              0.620000
23              21.990000             21.990000              0.000000
24              31.980000             33.990000             -2.010000
25              53.540000             59.490000             -5.950000
26              50.990000             50.990000              0.000000
27              19.590000             19.590000              0.000000
28              11.990000             11.990000              0.000000
29              26.110000             27.610000             -1.500000
30             145.230000            145.230000              0.000000
31               7.000000              9.990000             -2.990000
32              36.050000             37.670000             -1.620000
33              32.790000             32.790000              0.000000
34              54.440000             54.440000              0.000000
35              24.990000             24.990000              0.000000
36             157.750000            157.750000              0.000000
37              14.200000             14.200000              0.000000
38             330.390000            330.390000              0.000000
39              89.940000             89.940000              0.000000
40               6.670000              6.170000              0.500000
41              30.830000             26.990000              3.840000
42              59.650000             59.650000              0.000000
43              34.990000             34.990000              0.000000
44              17.620000             17.620000              0.000000
45              28.030000             27.950000              0.080000
46              17.490000             11.990000              5.500000
47            1028.190000           1023.370000              4.820000
48             105.200000            105.200000              0.000000
49              97.069388             99.348776             -2.279388
50              97.069388             99.348776             -2.279388

In [ ]:
mean_difference=final_df['DIFFERENCE_OF_PRICES'].mean()
mean_difference

-2.27938775510204

 If the value is positive, it indicates that Walmart prices are higher on average compared to Amazon prices for the same products. Conversely, if the value is negative, it suggests that Walmart prices are lower on average compared to Amazon prices.

In [ ]:
median_difference=final_df['DIFFERENCE_OF_PRICES'].median()
median_difference

0.0

If the median price difference is indeed zero, it may still be valuable to investigate further to understand the underlying factors contributing to this equilibrium in pricing between Walmart and Amazon for the products in the dataset.

In [ ]:
std_difference=final_df['DIFFERENCE_OF_PRICES'].std()
std_difference

9.750867532309782

A higher standard deviation indicates greater variability or dispersion in the price differences between Walmart and Amazon for the same products. It suggests that the price differences are more spread out from the mean difference.

Is there any correlation between Walmart prices and scores?

In [ ]:
correlation= df['WALMART_PRICE_CURRENT'].corr(df['SCORE'])
correlation

nan

If the correlation coefficient is NaN (Not a Number), it means that there is no linear relationship between the two variables or that one of the variables has no variability (e.g., all values are the same), resulting in a division by zero or other arithmetic error when calculating the correlation coefficient.

Are there any relationships between Walmart and Amazon product names or prices?

In [ ]:
corr = df['WALMART_PRICE_CURRENT'].corr(df['AMAZON_PRICE_CURRENT'])
corr

0.9978885471205412

 it indicates a strong positive linear relationship between the variables.

Do certain departments or categories tend to receive higher scores or have higher prices?

In [ ]:
dep=df.groupby('DEPARTMENT')[['SCORE','WALMART_PRICE_CURRENT']].mean()
dep

SCORE  WALMART_PRICE_CURRENT
DEPARTMENT                                                
Easter                          1.0              12.286667
Electronics                     1.0            1028.190000
Home, Furniture & Appliances    1.0              82.669974

from above departments have same score which is 1.0 , while electronics department has higher prices which is 1028.19.

In [ ]:
cat=df.groupby('CATEGORY')[['SCORE','WALMART_PRICE_CURRENT']].mean()
cat

SCORE  WALMART_PRICE_CURRENT
CATEGORY                                                  
Appliances                      1.0             330.390000
Bath                            1.0              22.760000
Bedding                         1.0              59.650000
Computers                       1.0            1028.190000
Decor                           1.0              46.422000
Easter Top Picks                1.0              12.286667
Furniture                       1.0             134.619231
Home                            1.0              99.826667
Home, Furniture & Appliances    1.0              97.069388
Kid's Rooms                     1.0             105.200000
Kitchen & Dining                1.0              40.542500
Storage & Organization          1.0              52.083333

from above,all the categories of walmart have same prices and computers have higher prices which is 1028.19 that is belongs to electronics department.

In [ ]:
dep=df.groupby('DEPARTMENT')[['SCORE','AMAZON_PRICE_CURRENT']].mean()
dep

SCORE  AMAZON_PRICE_CURRENT
DEPARTMENT                                               
Easter                          1.0             12.713333
Electronics                     1.0           1023.370000
Home, Furniture & Appliances    1.0             85.218671

Departments of amazon have same score which is 1.0 , and electronics has higher prices with 1023.37.

In [ ]:
cat=df.groupby('CATEGORY')[['SCORE','AMAZON_PRICE_CURRENT']].mean()
cat

SCORE  AMAZON_PRICE_CURRENT
CATEGORY                                                 
Appliances                      1.0            330.390000
Bath                            1.0             19.970000
Bedding                         1.0             59.650000
Computers                       1.0           1023.370000
Decor                           1.0             45.988000
Easter Top Picks                1.0             12.713333
Furniture                       1.0            138.624615
Home                            1.0            119.496667
Home, Furniture & Appliances    1.0             99.348776
Kid's Rooms                     1.0            105.200000
Kitchen & Dining                1.0             40.916250
Storage & Organization          1.0             53.930000

Categories of amazon have same score which is 1.0 , and computers has higher prices with 1023.37.

Is there a correlation between wallmart price and score?



In [ ]:
corr= df['WALMART_PRICE_CURRENT'].corr(df['SCORE']) # there is no linear relationship between both the variables.
corr

nan

there is no relationship between them

**Calculates the sum of Amazon and Walmart prices for each product and represent aggregated price information for the top two products listed on Amazon and Walmart?**

In [ ]:
df.groupby('AMAZON_NAME')['AMAZON_PRICE_CURRENT','WALMART_PRICE_CURRENT'].sum().reset_index().head(2)


<ipython-input-40-9225e1568391>:1: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  df.groupby('AMAZON_NAME')['AMAZON_PRICE_CURRENT','WALMART_PRICE_CURRENT'].sum().reset_index().head(2)


AMAZON_NAME  AMAZON_PRICE_CURRENT  \
0  10 Strawberry Street Simply White 36 Oz Pasta/...            220.687551   
1  10 Strawberry Street Two Tone Coupe 5.5?€?/24 ...             16.910000   

   WALMART_PRICE_CURRENT  
0             216.128776  
1              16.910000

by this above , we got to know about the total prices for each product of walmart and amazon and the aggregate price of the top two products of the these two amazon and walmart is 220.68 for named product (10 Strawberry Street Simply White 36 Oz Pasta/..) and 16.91 for (10 Strawberry Street Two Tone Coupe 5.5?€?/24) .

**Calculates various statistical measures (mean, median, max, min) for the current prices of products listed on Walmart within each group, and then resets the index of the resulting DataFrame?**

In [ ]:
df.groupby('WALMART_NAME').agg({'WALMART_PRICE_CURRENT':['mean', 'median','max','min']}).reset_index()
df.groupby('WALMART_NAME').agg({'AMAZON_PRICE_CURRENT':['mean', 'median','max','min']}).reset_index()

WALMART_NAME AMAZON_PRICE_CURRENT  \
                                                                      mean   
0   10 Strawberry Street Simply White 36 Oz Pasta/...            60.669388   
1   10 Strawberry Street Two Tone Coupe 5.5?€?/24 ...            16.910000   
2   10 Strawberry Street Two Tone Coupe 8.5?€?/36 ...            24.490000   
3   8" Brown Ivory and Orange Standing Duck Spring...            18.990000   
4   AcuRite Intelli-Time Digital Alarm Clock for B...            32.790000   
5   Alera ALEFW601416BL Compact 15.25 in. x 12.38 ...            27.610000   
6   Alera ALEFW601424BL 24 in. x 14 in. x 21 in. S...            37.670000   
7   Americanflat 12x18 Poster Frame in Silver - Co...            17.620000   
8   Americanflat 20x26 Puzzle Frame with Mat for 1...            34.990000   
9       Ameriwood Home Lamont Mantel Fireplace, White           330.390000   
10  Ameriwood Home Pursuit Mobile File Cabinet, Ch...           145.230000   
11  Arden Selections Outdoor Toss Pillow (2 Pack) ...            24.990000   
12               Boahaus Alva Dresser (Brown - White)           180.990000   
13  Caffoz 58" Home Entertainment Center Wood Medi...           142.720000   
14                     Coghlan's Featherweight Mirror             6.170000   
15       Contact US to Get 5000+ products comparisons            99.348776   
16  Convenience Concepts American Heritage Bookshe...           105.200000   
17  Convenience Concepts American Heritage Coffee ...           117.990000   
18  Convenience Concepts American Heritage Round C...           127.000000   
19  Convenience Concepts Designs2Go 4 Tier Corner ...            33.990000   
20  Convenience Concepts Designs2Go Highboy TV Sta...            89.000000   
21  Convenience Concepts Oxford Coffee Table with ...           105.580000   
22  Convenience Concepts Xtra Storage 5 Tier Foldi...            59.490000   
23         DHP Jett Junior Twin Metal Loft Bed, Black           150.000000   
24  Easter Bunny Wood Door Sign - Home Decor - 1 P...            15.570000   
25  Evriholder Lemon and Lime Squeezer, Yellow or ...             9.990000   
26       Furinno Simple Design Coffee Table, Espresso            29.770000   
27  Gallery Solutions 8x10 Black Wall or Tabletop ...            14.200000   
28  GreenPan Swift Healthy Ceramic Nonstick, 12 Pi...           134.990000   
29  Instapoints 5 Piece Gallery Wall Picture Frame...            54.440000   
30  Kate and Laurel Cates Farmhouse Wood Framed Wa...           157.750000   
31  Kinder Bueno Milk Chocolate and Hazelnut Cream...             3.580000   
32  KitchenAid Variable Speed Corded Hand Blender ...            50.990000   
33  Lavish Home 4-Tier Wood Folding Corner Display...            27.950000   
34  Lenovo ThinkCentre M70a Gen 3 11VL0040US Deskt...          1023.370000   
35  McCormick Air Fryer Loaded Seasoning Mix, 1.25 oz             1.490000   
36  Mind Reader 3 Tier Letter Tray Document Sorter...            19.590000   
37  Novogratz Kelly 5 Drawer Dresser, Faux Rattan ...           170.770000   
38        Novogratz Kelly Accent Cabinet, White/Taupe           133.750000   
39            Novogratz Voler Coffee Table, Light Oak           124.200000   
40  Ovente 8.5'' Hardwired Lighted Wall Mount Make...            89.940000   
41  Ovente Air Fryer 3.2 QT, 1400W with Adjustable...            55.980000   
42  Ovente French Coffee Press 12 Ounce Espresso M...            11.990000   
43  Patton Wall Decor 20" Black and Gold Metal Cut...            26.990000   
44  Prepac Space-Saving Entryway Organizer with Sh...           178.990000   
45  Sceptre 40" Class FHD (1080P) LED TV (X405BV-FSR)           207.000000   
46  Signature Design by Ashley Media Modern Geomet...            59.650000   
47  Smart Design Undershelf Storage Basket - Small...            11.990000   
48      South Shore Fusion 5-Drawer Chest, Pure White           199.990000   
49  South Shore Step One Essential 5-Drawer Chest,...          

This provides aggregated price statistics for each product listed on Walmart, including the mean, median, maximum, and minimum prices.

In [ ]:
#df.groupby('AMAZON_NAME')['AMAZON_PRICE_CURRENT'].sum().reset_index().sort_values(by='AMAZON_PRICE_CURRENT')
